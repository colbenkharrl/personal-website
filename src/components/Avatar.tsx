import clsx from 'clsx';
import Image from 'next/image';
import Link from 'next/link';
import avatarImage from '@/images/avatar_512.jpg';

export function Avatar({
  large = false,
  className,
  ...props
}: Omit<React.ComponentPropsWithoutRef<typeof Link>, 'href'> & {
  large?: boolean;
}) {
  return (
    <Link
      href="/"
      aria-label="Home"
      className={clsx(className, 'pointer-events-auto')}
      {...props}
    >
      <Image
        src={avatarImage}
        alt=""
        sizes={large ? '4rem' : '2.25rem'}
        className={clsx(
          'rounded-full bg-zinc-100 object-cover dark:bg-zinc-800',
          large ? 'h-16 w-16' : 'h-9 w-9',
        )}
        priority
      />
    </Link>
  );
}
