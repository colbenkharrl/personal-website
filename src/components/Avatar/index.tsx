import clsx from 'clsx';
import Image from 'next/image';
import Link from 'next/link';
import avatarImage from '@/images/avatar_512.jpg';

export type AvatarProps = Omit<
  React.ComponentPropsWithoutRef<typeof Link>,
  'href'
> & {
  /**
   * Toggle for large button styles.
   */
  large?: boolean;
};

export function Avatar({ large = false, className, ...props }: AvatarProps) {
  return (
    <Link
      href="/"
      aria-label="Home"
      className={clsx(className, 'pointer-events-auto')}
      {...props}
    >
      <Image
        src={avatarImage}
        alt=""
        sizes={large ? '4rem' : '2.25rem'}
        className={clsx(
          'rounded-full bg-zinc-100 object-cover dark:bg-zinc-800',
          large ? 'h-16 w-16' : 'h-9 w-9',
        )}
        priority
      />
    </Link>
  );
}
