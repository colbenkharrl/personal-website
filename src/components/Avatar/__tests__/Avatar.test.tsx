import { render, screen } from '@testing-library/react';
import * as stories from '../__stories__/Avatar.stories';
import { composeStories } from '@storybook/react';

const { Basic, Large } = composeStories(stories);

it('Avatar matches snapshot (regression test)', () => {
  render(
    <div data-testid="Avatar">
      <Basic />
      <Large />
    </div>,
  );
  expect(screen.getByTestId('Avatar')).toMatchSnapshot(
    'Full component snapshot.',
  );
});
