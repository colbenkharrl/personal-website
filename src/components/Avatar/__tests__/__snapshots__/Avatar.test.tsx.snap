// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar matches snapshot (regression test): Full component snapshot. 1`] = `
<div
  data-testid="Avatar"
>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <a
    aria-label="Home"
    class="pointer-events-auto"
    href="/"
  >
    <img
      alt=""
      class="rounded-full bg-zinc-100 object-cover dark:bg-zinc-800 h-9 w-9"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="40"
      sizes="2.25rem"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
      style="color: transparent;"
      width="40"
    />
  </a>
  <script>
    !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&true)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}if(e==='light'||e==='dark')d.style.colorScheme=e}catch(e){}}()
  </script>
  <a
    aria-label="Home"
    class="pointer-events-auto"
    href="/"
  >
    <img
      alt=""
      class="rounded-full bg-zinc-100 object-cover dark:bg-zinc-800 h-16 w-16"
      data-nimg="1"
      decoding="async"
      fetchpriority="high"
      height="40"
      sizes="4rem"
      src="/_next/image?url=%2Fimg.jpg&w=3840&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=16&q=75 16w, /_next/image?url=%2Fimg.jpg&w=32&q=75 32w, /_next/image?url=%2Fimg.jpg&w=48&q=75 48w, /_next/image?url=%2Fimg.jpg&w=64&q=75 64w, /_next/image?url=%2Fimg.jpg&w=96&q=75 96w, /_next/image?url=%2Fimg.jpg&w=128&q=75 128w, /_next/image?url=%2Fimg.jpg&w=256&q=75 256w, /_next/image?url=%2Fimg.jpg&w=384&q=75 384w, /_next/image?url=%2Fimg.jpg&w=640&q=75 640w, /_next/image?url=%2Fimg.jpg&w=750&q=75 750w, /_next/image?url=%2Fimg.jpg&w=828&q=75 828w, /_next/image?url=%2Fimg.jpg&w=1080&q=75 1080w, /_next/image?url=%2Fimg.jpg&w=1200&q=75 1200w, /_next/image?url=%2Fimg.jpg&w=1920&q=75 1920w, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2048w, /_next/image?url=%2Fimg.jpg&w=3840&q=75 3840w"
      style="color: transparent;"
      width="40"
    />
  </a>
</div>
`;
